### ISSUE TITLE
Usefulness of property Enumeration_Obsolete

### ISSUE THREAD

**Issue by emopers** [runtimeverification/javamop#234]
Hi we're thinking of the usefulness of property [Enumeration_Obsolete](https://runtimeverification.com/monitor/annotated-java/__properties/html/java/util/Enumeration_Obsolete.html). First in Java API, it just stated that "using Iterator in preference to Enumeration", which seems not to be officially depreciated as obsolete. Furthermore, StringTokenizer is the sole implementation of this interface in JDK and widely used by developers now. 
From our observation this property triggered large amount of violations when running JavaMOP over many open source projects. So we'd suggest that this property could at least, just check for the instances implementing Enumeration that are not StringTokenizer. This would help to eliminate the noise generated during monitoring.

**Comment by xiaohe27** [comment]
@emopers , I agree with your arguments. You can modify the behavior of monitoring by improving the mop spec, specifically the abstraction level of the point cuts. Currently, it is defined as "Enumeration+.*(..)", so every method with that prefix will generate an abstract "use" event. We can make it more accurate by providing more fine-grained, e.g. conditional point cuts, so the events will be generated only if the method receiver is not an instance of StringTokenizer. Let me know if you need further help.
